Formula/pnpm.rb

```diff
@@ -20,14 +20,65 @@ class Pnpm < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "7da2aca6833071eec6626c22e5026af7587cd297ce324d49b4f3bdffb60ea8ee"
   end
 
-  depends_on "node"
+  depends_on "node@14" => :build
+
+  # Described in https://github.com/pnpm/pnpm#installation
+  # Managed by https://github.com/pnpm/get
+  resource "pnpm-buildtime" do
+    url "https://get.pnpm.io/v6.7.js"
+    sha256 "56bffe269aab055bec3a07fa65c387f5a35729550dc7fc76711844a2de6e1b8b"
+  end
 
   def install
-    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
-    bin.install_symlink Dir["#{libexec}/bin/*"]
+    buildtime_bin = buildpath/"buildtime-bin"
+    resource("pnpm-buildtime").stage do |r|
+      buildtime_bin.install "v#{r.version}.js"
+      (buildtime_bin/"pnpm").write_env_script "#{Formula["node@14"].bin}/node",
+        buildpath/"buildtime-bin/v#{r.version}.js", {}
+      chmod 0755, buildtime_bin/"pnpm"
+    end
+    ENV.prepend_path "PATH", buildtime_bin
+    system "pnpm", "install"
+    chdir "packages/pnpm" do
+      system buildpath/"node_modules/.bin/tsc", "--build"
+      system "pnpm", "run", "bundle"
+      on_macos do
+        system "node_modules/.bin/pkg", "--target=host", "--out-path=dist", "dist/pnpm.cjs"
+      end
+      on_linux do
+        system "node_modules/.bin/pkg", "--target=linuxstatic", "--out-path=dist", "dist/pnpm.cjs"
+      end
+      bin.install "dist/pnpm"
+    end
+  end
+
+  def caveats
+    <<~EOS
+      You should create npm's global directory if you want to install packages globally:
+
+        mkdir -p ~/npm-global/bin
```

Comment:
This directory may be created anywhere, the only requirement is for the directory to be under "pnpm" or ".pnpm" directory. And for the directory to be in PATH

Reply:
Reworded.